In [1]:
import numpy as np
import pandas as pd

In [2]:
data = pd.DataFrame({
    'age' :     [ 10, 22, 13, 21, 12, 11, 17],
    'section' : [ 'A', 'B', 'C', 'D', 'E', 'F', 'G'],
    'country' : [ 'USA', 'GBR', 'TUR', 'NLD', 'BEL', 'DEU', 'FRA'],
    'gender' :  [ 'M', 'F', 'F', 'M', 'M', 'M', 'F'],
    'favourite_color' : [ 'red', 'blue', 'yellow', 'pink', 'black', 'green', 'red']
})
data.set_index('section', inplace=True)
data

age country gender favourite_color
section                                    
A         10     USA      M             red
B         22     GBR      F            blue
C         13     TUR      F          yellow
D         21     NLD      M            pink
E         12     BEL      M           black
F         11     DEU      M           green
G         17     FRA      F             red

he DataFrame data has already been created above. How to select the rows where ages are greater than 15 as below? 

In [3]:
data[data['age'] > 15]

age country gender favourite_color
section                                    
B         22     GBR      F            blue
D         21     NLD      M            pink
G         17     FRA      F             red

The DataFrame data has already been created above. How to select the country column where ages are greater than 15 as below? 

In [4]:
data[data['age']>15]['country']

section
B    GBR
D    NLD
G    FRA
Name: country, dtype: object

The DataFrame data has already been created above. How to select the rows where the favourite_colurs are red and the ages are greater than 15 as below? 

In [5]:
data[(data['age']>15) & (data['favourite_color'] == 'red')]

age country gender favourite_color
section                                    
G         17     FRA      F             red

In [6]:
data[(data['age']>20) | (data['favourite_color'] == 'red')]

age country gender favourite_color
section                                    
A         10     USA      M             red
B         22     GBR      F            blue
D         21     NLD      M            pink
G         17     FRA      F             red

In [7]:
outside = ['A1','A1','A1','A2','A2','A2']
inside = [1,2,3,1,2,3]
hier_index = list(zip(outside,inside))
hier_index = pd.MultiIndex.from_tuples(hier_index)
np.random.seed(101)
data = pd.DataFrame(np.random.randn(6,2),index=hier_index,columns=['X','Y'])
data

X         Y
A1 1  2.706850  0.628133
   2  0.907969  0.503826
   3  0.651118 -0.319318
A2 1 -0.848077  0.605965
   2 -2.018168  0.740122
   3  0.528813 -0.589001

The Multi-Indexed DataFrame data is given above. How to get the following output?
<br>1    0.605965
<br>2    0.740122
<br>3   -0.589001

In [8]:
data['Y']['A2']

1    0.605965
2    0.740122
3   -0.589001
Name: Y, dtype: float64

In [9]:
data.loc['A1','X'] == data['X']['A1'] 

1    True
2    True
3    True
Name: X, dtype: bool

The Multi-Indexed DataFrame data is given above. How to get the following output?
<br>0.605965

In [10]:
data.loc['A2'].loc[1, 'Y']

0.6059653494949336